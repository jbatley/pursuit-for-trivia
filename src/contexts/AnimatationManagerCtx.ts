import React from 'react';

export interface AnimatationManager {
  animator: boolean;
}

const AnimatationManagerCtx = React.createContext<AnimatationManager>({
  animator: false,
});

export default AnimatationManagerCtx;
