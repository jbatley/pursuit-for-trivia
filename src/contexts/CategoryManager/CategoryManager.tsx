import React, {
  useState, createContext, useContext, useEffect,
} from 'react';
import LS from 'utils/storage';
import config from 'config';
import { fetchCategories } from 'api';

interface Props {
  children?: React.ReactChild;
}

interface CategoryManager {
  categories: CategoriesArray | null | undefined;
  selected: number | null;
  setCategory: React.Dispatch<number>;
  difficulty: QuestionDifficulty | undefined;
  setDifficulty: React.Dispatch<QuestionDifficulty | undefined>;
}

let ls = new LS(config.lsKeys.categories);
let isInStorage = ls.get();

export const CategoryManagerCtx = createContext<CategoryManager | void>(undefined);

export function useCategoryManager(): CategoryManager {
  let context = useContext(CategoryManagerCtx);
  if (context === undefined) {
    throw new Error('useCategoryManager must be used within a CategoryManagerProvider');
  }

  return context;
}

export const CategoryManagerProvider: React.FC<Props> = ({ children }: Props) => {
  let [categories, setCategories] = useState<CategoriesArray | null | undefined>(undefined);
  let [selected, setSelected] = useState<number | null>(null);
  let [difficulty, setDifficulty] = useState<QuestionDifficulty | undefined>(undefined);

  useEffect(() => {
    async function loadCategories() {
      if (isInStorage !== null) {
        setCategories(isInStorage as CategoriesArray);
      } else {
        try {
          let data = await fetchCategories();
          setCategories(data?.trivia_categories);
          ls.set(data?.trivia_categories);
        } catch {
          // PROPER ERRO
          setCategories(null);
        }
      }
    }
    if (categories === undefined) {
      loadCategories();
    }
  }, [categories]);

  let values = {
    categories,
    selected,
    setCategory: setSelected,
    difficulty,
    setDifficulty,
  };

  return (
    <CategoryManagerCtx.Provider value={values}>
      {children}
    </CategoryManagerCtx.Provider>
  );
};
