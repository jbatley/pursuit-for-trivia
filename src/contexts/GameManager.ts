import React from 'react';

export interface GameManager {
  timer: number;
  lives: number;
  resetTimer: () => void;
  cancelTimer: () => void;
  startTimer: () => void;
  loseLife: () => void;
  resetLife: () => void;
  timesUp: () => void;
  correct: () => void;
  incorrect: () => void;
}

const GameManagerCtx = React.createContext<GameManager>({
  timer: 0,
  lives: 0,
  resetTimer: () => {},
  cancelTimer: () => {},
  startTimer: () => {},
  loseLife: () => {},
  resetLife: () => {},
  timesUp: () => {},
  correct: () => {},
  incorrect: () => {},
});

export default GameManagerCtx;
