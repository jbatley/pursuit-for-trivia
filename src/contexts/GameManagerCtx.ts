import React from 'react';

export interface GameManager {
  lives: number;
  timesUp: () => void;
  resetGame: () => void;
  startGame: (mode: string) => void;
  correct: () => Promise<void>;
  incorrect: () => Promise<void>;
}

const GameManagerCtx = React.createContext<GameManager>({
  lives: 0,
  timesUp: () => {},
  resetGame: () => {},
  startGame: () => {},
  correct: async () => {},
  incorrect: async () => {},
});

export default GameManagerCtx;
