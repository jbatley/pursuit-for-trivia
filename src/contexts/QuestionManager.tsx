import { createContext } from 'react';

export interface QuestionManager {
  current: number;
  next: () => void;
  reset: () => void;
  fetch: () => Promise<void>;
  questionArray: Question[] | null;
  shuffle: () => void;
  clear: () => void;
}

const QuestionManagerCtx = createContext<QuestionManager>({
  current: 0,
  next: () => {},
  fetch: async () => {},
  reset: () => {},
  questionArray: null,
  shuffle: () => {},
  clear: () => {},
});

export default QuestionManagerCtx;
