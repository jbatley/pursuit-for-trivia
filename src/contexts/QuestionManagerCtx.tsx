import { createContext } from 'react';

export interface QuestionManager {
  current: number;
  nextQuestion: () => Promise<void>;
  reset: () => void;
  fetch: () => Promise<void>;
  questionArray: Question[] | null;
  revealAnswers: () => void;
  shuffle: () => void;
  clear: () => void;
}

const QuestionManagerCtx = createContext<QuestionManager>({
  current: 0,
  nextQuestion: async () => {},
  fetch: async () => {},
  reset: () => {},
  questionArray: null,
  revealAnswers: () => {},
  shuffle: () => {},
  clear: () => {},
});

export default QuestionManagerCtx;
