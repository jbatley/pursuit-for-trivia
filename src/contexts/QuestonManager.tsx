import { createContext } from 'react';

const QuestionManagerContext = createContext({
  a: () => {},
});

export default QuestionManagerContext;
