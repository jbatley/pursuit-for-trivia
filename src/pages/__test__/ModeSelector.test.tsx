import React from 'react';
import { render as testRender } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import { CategoryManagerProvider } from 'contexts/CategoryManager';

import Component from '../ModeSelector';

function render(children: React.ReactChild) {
  return testRender(
    <Router>
      <CategoryManagerProvider>
        {children}
      </CategoryManagerProvider>
    </Router>,
  );
}

beforeEach(() => {
  jest.clearAllMocks();
});

describe('Page ModeSelector', () => {
  test('should render and', async () => {
    const { getByText } = render(<Component />);
    expect(getByText(/set up/i)).toBeInTheDocument();
  });
  test('matches snapshot', async () => {
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
