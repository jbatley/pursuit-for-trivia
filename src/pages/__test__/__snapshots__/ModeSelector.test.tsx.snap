// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page ModeSelector matches snapshot 1`] = `
<section
  class="grid"
  style="grid-template-rows: 10% 40% 40% 10%; padding: 25px;"
>
  <h1
    class="h1"
  >
    Set up
  </h1>
  <div
    class="flex"
  >
    <h3
      class="container"
    >
      How to play
    </h3>
    <span
      class="text"
    >
      
          You get 10 seconds to answer a question, the quicker you are the more pointes you.

          If you're wrong, you lose a life and lose all 3 lives then GAME OVER.
          
    </span>
  </div>
  <div
    class="flex"
  >
    <label
      class="label"
    >
      Category
      <select
        class="select"
        data-testid="dropdown"
      >
        <option
          disabled=""
          selected=""
          value="Symbol(selected)"
        >
          Category
        </option>
        <option
          value=""
        >
          Any
        </option>
      </select>
    </label>
    <label
      class="label"
    >
      Difficulty
      <select
        class="select"
        data-testid="dropdown"
      >
        <option
          disabled=""
          selected=""
          value="Symbol(selected)"
        >
          Difficulty
        </option>
        <option
          value=""
        >
          Any
        </option>
        <option
          value="easy"
        >
          easy
        </option>
        <option
          value="medium"
        >
          medium
        </option>
        <option
          value="hard"
        >
          hard
        </option>
      </select>
    </label>
  </div>
  <a
    class="link"
    href="/game/normal"
  >
    <button
      class="box"
      type="button"
    >
      Play
    </button>
  </a>
</section>
`;
